src/helpers/tree.js

```diff
@@ -0,0 +1,48 @@
+/*eslint func-style:0*/
+import mapObj from './mapObj';
+import values from './values';
+
+function map(fn) {
+    return function (data) {
+        if (!data) return;
+
+        if (data.map) return data.map(fn);
+
+        if (typeof data === 'object') {
+            return mapObj(fn, data);
+        }
+    }
+}
+
+function extract(obj) {
+    return obj.extract();
+}
+
+export default function tree({ ref, refs }) {
```

Comment:
@kaw2k I really like how compact these methods are. My one comment would be, for readability / digestibility purposes, it would probably be nice to have signatures above these methods.